include/gpio.h

```diff
@@ -108,18 +108,11 @@ extern "C" {
 #define GPIO_PUD_MASK		(3 << GPIO_PUD_POS)
 /** @endcond */
 
-/*
- * GPIO_PIN_(EN-/DIS-)ABLE are for pin enable / disable.
- *
- * Individual pins can be enabled or disabled
- * if the controller supports this operation.
- */
-
-/** Enable GPIO pin. */
-#define GPIO_PIN_ENABLE		(1 << 10)
+/** Deprecated, do not use - Enable GPIO pin. */
+#define GPIO_PIN_ENABLE		(0)
 
-/** Disable GPIO pin. */
-#define GPIO_PIN_DISABLE	(1 << 11)
+/** Deprecated, do not use - Disable GPIO pin. */
+#define GPIO_PIN_DISABLE	(0)
```

Comment:
Deprecated macros should be marked with `__DEPRECATED_MACRO`.

Reply:
Thanks! I have an extra question though. The `__DEPRECATED_MACRO` seems to be unused and I miss an example. If I do
```
#define GPIO_PIN_ENABLE                (0) __DEPRECATED_MACRO
```
checkpatch complains
```
-:10: ERROR:COMPLEX_MACRO: Macros with complex values should be enclosed in parentheses
#10: FILE: include/gpio.h:112:
+#define GPIO_PIN_ENABLE                (0) __DEPRECATED_MACRO

```
This one looks a bit strange but seems to work
```
#define GPIO_PIN_ENABLE       (0 __DEPRECATED_MACRO)
```
Is it how the macro is supposed to be used?

Reply:
checkpatch.pl is really, really stupid unfortunately

Reply:
have you tried #define GPIO_PIN_ENABLE __DEPRECATED_MACRO 0

I don't see why you have to put the 0 in parens, and it might make the preprocessor thing its an argument list

Reply:
Unfortunately that doesn't work, checkpatch still complains. The only one which checkpatch seems to be fine with is
```
#define GPIO_PIN_ENABLE       (0 __DEPRECATED_MACRO)
```
Should I push it or do we want to fix checkpatch?

Reply:
push something that works and that is valid, do not worry about checkpatch for this

Reply:
Fixed.